### PR TITLE
231 write sphinx documentation using markdown via myst

### LINKED ISSUE #231: Write Sphinx documentation using Markdown via MyST
> 2. Do we want to keep using rst? [I Sphinx supports markdown](https://www.sphinx-doc.org/en/master/usage/markdown.html) (see [here](https://myst-parser.readthedocs.io/en/latest/) also). Maybe, a hybrid between the two would be a solution?

If you feel strongly that this is the better way to go, please open an issue and describe the "hybrid" solution a bit more. If you are on the fence, I think we can just continue with the current workflow and keep focused on higher-priority tasks.

_Originally posted by @damonbayer in https://github.com/CDCgov/multisignal-epi-inference/issues/224#issuecomment-2200413117_

### LINKED ISSUE #414: Deterministic module documentation missing from the webpage

### PR DESCRIPTION
**Pull request by dylanhmorris**
This PR:

- switches all documentation source from `.rst` to `.md`
- updates sphinx settings and github actions accordingly
- Removes no longer needed `post_process_rst.py`
- replaces MathJax with KaTeX for equation rendering
- Fixes #414 
- Removes some legacy files

Closes #231

### CODE REVIEW

**Comment by dylanhmorris** [comment]
As of ac5ac13 this will also close #414

**Comment by dylanhmorris** [comment]
This should also resolve #413

File: docs/source/developer_documentation.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+- Multidimensional array conventions
+
+  - In a multidimensional array of timeseries, time is always the first dimension. By default, `jax.lax.scan()` and `numpyro.contrib.control_flow.scan()` build output arrays by augmenting the first dimension, and variables are often scanned over time, making default output of scan over time sensible.
+
+## Adding Documentation to Sphinx
+
+The project uses Sphinx to generate documentation. To learn more about Sphinx, visit the [Sphinx documentation](https://www.sphinx-doc.org/en/master/).
+
+### New module or classes
```
Comment (damonbayer) [inline docs/source/developer_documentation.md:53]: This section should be updated with markdown examples.

File: docs/source/developer_documentation.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+This entry tells Sphinx to generate documentation for the `HospitalAdmissions` module and its members.
+
+### New tutorials
+
+`PyRenew` tutorials are [quarto documents](https://quarto.org) located under [./docs/source/tutorials](https://github.com/CDCgov/PyRenew/tree/main/docs/source/tutorials). Tutorials are automatically rendered using GitHub actions (see the Workflow file [here](https://github.com/CDCgov/PyRenew/actions/workflows/website.yaml)).
+
+To make the new tutorial available in the website, developers should follow these steps:
+
+1. Create a new `quarto` file in the `./docs/source/tutorials` directory. For instance, the `example_with_datasets.qmd` file was added to the repository.
```
Comment (damonbayer) [inline docs/source/developer_documentation.md:77]: example_with_datasets Example is out of date.

File: docs/source/developer_documentation.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+:maxdepth: 2
+../index
+../pyrenew_reference/index
+../tutorials/index
+../genindex
+../developer_documentation
+```
+````
+
+The last entry is the `developer_documentation` page.
```
Comment (damonbayer) [inline docs/source/developer_documentation.md:110]: Why this comment?

**Review by damonbayer — APPROVED** [review]
Thanks @dylanhmorris. I think a but more cleanup could be done before or after mergin this. See my comments.

**Comment by dylanhmorris** [comment]
> Thanks @dylanhmorris. I think a but more cleanup could be done before or after mergin this. See my comments.

I think after makes sense for atomicity

**Pull request merged**